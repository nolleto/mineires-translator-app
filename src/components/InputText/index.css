.input-text {
  display: flex;
  flex-direction: column;
  align-content: center;
  text-align: start;
  margin: 10px;
}

.input-text__span {
  margin-bottom: 10px;
  font-size: 18px;
  font-weight: bold;
}

.input-text__textarea {
  min-height: 100px;
  font-size: 18px;
  border-radius: 3px;
}

.input-text__textarea:focus {
  outline: none !important;
  border:1px solid #e74c3c;
  box-shadow: 0 0 10px #e74c3c;
}
