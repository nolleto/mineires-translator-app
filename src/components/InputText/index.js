import './index.css';

import PropTypes from 'prop-types';
import React from 'react';

const InputText = ({ text, setText }) => {
  return (
    <div className='input-text'>
      <span className='input-text__span'>
        Texto em PortuguÃªs
      </span>

      <textarea
        className='input-text__textarea'
        value={text}
        onChange={({ target }) => setText(target.value)}
      />
    </div>
  )
};

InputText.propTypes = {
  text: PropTypes.string,
  setText: PropTypes.func.isRequired
};

export default InputText;
