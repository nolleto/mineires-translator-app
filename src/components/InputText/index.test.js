import InputText from './index';
import React from 'react';
import { shallow } from 'enzyme';

const shallowComponent = (props = {}) => {
  const allProps = {
    setText: jest.fn(),
    ...props
  };

  return shallow(
    <InputText
      {...allProps}
    />
  );
};

describe('InputText component', () => {
  describe('Snapshot', () => {
    it('when has no text', () => {
      const snapshot = shallowComponent();

      expect(snapshot).toMatchSnapshot();
    });

    it('when has text', () => {
      const snapshot = shallowComponent({ text: 'some text' });

      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Event Listeners', () => {
    it('when textarea changes calls "setText" with the new text', () => {
      const setText = jest.fn();
      const value = 'new text';
      const component = shallowComponent({ setText });
      const textarea = component.find('textarea');
      textarea.simulate('change', { target: { value } });

      expect(setText).toBeCalledWith(value);
    });
  });
});
