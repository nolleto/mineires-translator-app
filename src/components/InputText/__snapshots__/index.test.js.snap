// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText component Snapshot when has no text 1`] = `
<div
  className="input-text"
>
  <span
    className="input-text__span"
  >
    Texto em Português
  </span>
  <textarea
    className="input-text__textarea"
    onChange={[Function]}
  />
</div>
`;

exports[`InputText component Snapshot when has text 1`] = `
<div
  className="input-text"
>
  <span
    className="input-text__span"
  >
    Texto em Português
  </span>
  <textarea
    className="input-text__textarea"
    onChange={[Function]}
    value="some text"
  />
</div>
`;
